### DISCUSSION TITLE
Possible to access an OAuth endpoint without a frontend for my service?

### DISCUSSION THREAD

**Discussion by Brettyoke49** [etsy/open-api#910] [Q&A] [answered]
Hello,

I'm writing a background service that generates a nice list of information for my shop based on the results of the `getListingsByShop` endpoint. This endpoint, however, requires an OAuth2 authentication header. The instructions in the [official documentation](https://developers.etsy.com/documentation/essentials/authentication/) seem aimed towards applications with a front-end where users can sign in to their shop.

My application runs on an AWS server with no front-end at all and only does work for ONE shop. Is it possible to get OAuth for that one shop/login solely on the server side of things? I am more than happy to do something hacky.

**Comment by PMStanley** [comment, marked as answer]
A very quick and dirty way would be to just follow the [Quick Start tutorial](https://developers.etsy.com/documentation/tutorials/quickstart) to set up a local front-end to grab your tokens and copy them manually to the server app.  For that point onwards you'd be able to use the tokens programatically and wouldn't require a persistent front-end.

The caveat here is you need your app approving for OAuth access by Etsy before you can do what you're planning.

**Reply by Brettyoke49** [reply]
Is OAuth Access the same as commercial access? That feels silly if I only want OAuth for one person (me)

**Reply by PMStanley** [reply]
> Is OAuth Access the same as commercial access? That feels silly if I only want OAuth for one person (me)

It's been while but if I remember correctly once you've created an app in the developer portal you can request additional access (to get OAuth enabled) without requiring commercial access to be granted.  

For me it was a 3 stage affair: 1) create app and get API key, 2) request additional access for OAuth, 3) (much later) request commercial access so other users can use my app.

Obviously you don't need to take step 3, though you will need to explain why you're requesting additional access in step 2

**Reply by showtroylove** [reply]
@PMStanley you mention additional OAuth access above.  Can you tell me what this additional access is?  There is a list of scopes published on api.etsy.com.  Are there other capabilities beyond that?  Please advise.

**Reply by PMStanley** [reply]
@showtroylove it’s not a scope. When I first created my app on the dev portal I got a key but no OAuth access. I had to specifically request it by completing the app form and wait for approval before I could call any endpoints that require OAuth

**Reply by JohnWinner** [reply]
I had the same problem and came here to realize that ... yes indeed, it is THAT complicated just to get a token for myself and me alone. Jeez! I know other API (EbayPartnerNetwork) that don't require you to write a frontend just to receive a token! It is not developer friendly.
